crates/oxc_transformer/src/visit_utils.rs

```diff
@@ -0,0 +1,61 @@
+use oxc_ast::ast::{Expression, Statement};
+use oxc_ast::VisitResult;
+
+#[derive(Default)]
+pub struct TransformResult<'a> {
```

Comment:
I think we should make this into an enum for example

```rust
enum TransformerResult<'a> {
    Pass,
    Append(Vec<'a, TransformReplaceKind<'a>>,
    Replace(Vec<'a, TransformReplaceKind<'a>>,
}
```

Reply:
Maybe append isn't needed at all, Since replacing something with a list containing that same node would cost us cloning we might consider adding something like append if we find a usecase for it